### DISCUSSION TITLE
Adding CAA record to Vercel-managed subdomains

### DISCUSSION THREAD

**Discussion by torablien** [vercel/community#4955] [Help] [answered]
### Summary

I am trying to add Amazon as a `CAA` to my subdomains that have `cname.vercel-dns.com.` as their `CNAME`.  My apex domain is configured to Vercel with an `A` record so I am able to just add the Amazon `CAA` record for the apex domain in my registrar's DNS settings.

I've added the following entries to the domain's DNS records in Vercel (Domains -> DNS Records):
| Name | Type | Value                                     | TTL |
| ---- | ---- | ----------------------------------------- | --- |
| dev  | CAA  | 0 issue "[amazon.com](http://amazon.com)" | 60  |
|      | CAA  | 0 issue "[amazon.com](http://amazon.com)" | 60  |

but when I check the records for any of my subdomains (including `dev.`) like with a [DNS Checker](https://www.whatsmydns.net/#CAA/), I only see:
```
0 issue "globalsign.com"
0 issue "letsencrypt.org"
0 issue "sectigo.com"
```
and not the Amazon one.

In case it helps, when I add a `CAA` record in Vercel, it automatically also adds a `0 issue "letsencrypt.org"` record but interestingly not the globalsign or sectigo ones, despite them showing on DNS Checker. The subdomains are Preview domains that are linked to Git branches (and otherwise working as expected). 

Is there anything else I need to do to add the CAA record?

### Steps to Reproduce

On https://vercel.com/dashboard/domains, add a CAA record to a Vercel managed domain/subdomain and then check https://www.whatsmydns.net/#CAA/ to see if it updates.

**Comment by torablien** [comment, marked as answer]
I came across this note, https://vercel.com/guides/can-a-caa-record-be-added-to-a-subdomain-configured-with-cname, so I tried removing the `CNAME` records for my subdomains and instead using `A` records [76.76.21.21](https://vercel.com/guides/a-record-and-caa-with-vercel) (despite the only visible options in Vercel being `CNAME` or `NS`). I also deleted the DNS zone completely in Vercel and just added a letsencrypt.org `CAA` (apex) record in my registrar's DNS settings.

This seems to be working and lets me control the CAA records from my domain registrar. Are there any issues with this approach?

**Reply by amyegan** [reply]
CAA records cannot be used alongside CNAME records, so using A records is the right approach. 

For example, if your domain was `example.com` then you would
1. Create an A record for `www.example.com` with the value `76.76.21.21`
2. Create a CAA record for `www.example.com` with the value `0 issue "letsencrypt.org"` to allow Vercel to provision a SSL certificate for you. _(Note: Vercel provides all domains with custom SSL certificates. Only Enterprise teams can upload their own SSL certificate.)_

**Reply by torablien** [reply]
Thanks!

**Reply by Lucas-Bonafe** [reply]
>     1. Create an A record for `www.example.com` with the value `76.76.21.21`
> 
>     2. Create a CAA record for `www.example.com` with the value `0 issue "letsencrypt.org"` to allow Vercel to provision a SSL certificate for you. _(Note: Vercel provides all domains with custom SSL certificates. Only Enterprise teams can upload their own SSL certificate.)_

@amyegan, when configuring subdomains, do I just need to add an A record with the name 'subdomain' pointing to 76.76.21.21?

My main domain was configured with CNAME, I simply delete it to avoid conflicts and with these 3 records I already have a Domain and Subdomain configured in different projects?
